tests/parsing.py

```diff
@@ -0,0 +1,58 @@
+# Small helper class to check the correct parsing
+
+from lark import Lark
+from lark.lexer import Token
+from typing import List, Tuple, Optional
+
+
+class ParsingCheck:
+
+    def __init__(self, lark):
+        """\
+        Args:
+            lark: path of a lark grammar file.
+        """
+        self.parser = Lark(open(lark), parser="lalr")
+
+
+    def precedence_check(self, formula: str, tokens: List[str],
+            print_tree: bool = False) -> Tuple[bool, Optional[str]]:
+        """\
+        Parse the formula and check that operators have the expected
+        precedence. 
+
+        Args:
+            formula: expression to parse
+            tokens: a list of symbols (top-down, left-right) order in the
+                parsing tree
+            print_tree: verbose parsing tree
+        Returns:
+            false, if parsing did not respect the order; and a string of error
+        """
```

Comment:
We use reStructuredText Docstring format, as recommended by PEP 287: https://www.python.org/dev/peps/pep-0287/

However, let's discuss what best fits our need. But eventually, we should stick with only one format across the codebase.

Reply:
Sure! Thanks for the feedback. Let me know for other style differences.
Essentially, apart from text markup, I should write: `:param my_arg: comment`, right?
Also, should I avoid commenting the arguments? Maybe it's too verbose, in presence of type hints.

Reply:
- yes, but not only `:param:` but also `:return:` and optionally `:raises:`. I think these are the main tags.
- No I think it is great to comment parameters as well. Some documentation tools will read the docstrings and generate the API docs based on that.

As an example, have a look a this:
https://github.com/whitemech/pythomata/blob/master/pythomata/core.py#L147-L159

I tried the docstring with all the three tags mentioned. But of course have a look online.